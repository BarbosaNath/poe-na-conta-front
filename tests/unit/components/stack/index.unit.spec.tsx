import React from 'react';
import Stack from '@components/stack';
import { render } from '@testing-library/react';

describe('Stack', () => {
  it('should render with default props', () => {
    const { container } = render(<Stack>Test</Stack>);

    expect(container).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const { container } = render(
      <Stack
        space="sm"
        align="center"
        justify="end"
        direction="row"
        wrap="wrap"
      >
        Test
      </Stack>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with wrap = true', () => {
    const { container } = render(<Stack wrap>Test</Stack>);

    expect(container).toMatchSnapshot();
  });

  it('should render with fullWidth = true', () => {
    const { container } = render(<Stack fullWidth>Test</Stack>);

    expect(container).toMatchSnapshot();
  });

  it('should render with custom element', () => {
    const { container } = render(<Stack as="main">Test</Stack>);

    expect(container).toMatchSnapshot();
  });
});
