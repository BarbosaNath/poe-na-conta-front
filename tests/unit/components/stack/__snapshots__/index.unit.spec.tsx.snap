// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stack > should render with custom element 1`] = `
<div>
  <main
    class="lds--stack lds--stack--nowrap lds--stack--space-lg lds--stack--direction--column lds--stack--align-start lds--stack--justify-start"
  >
    Test
  </main>
</div>
`;

exports[`Stack > should render with custom props 1`] = `
<div>
  <div
    class="lds--stack lds--stack--wrap lds--stack--space-sm lds--stack--direction--row lds--stack--align-center lds--stack--justify-end"
  >
    Test
  </div>
</div>
`;

exports[`Stack > should render with default props 1`] = `
<div>
  <div
    class="lds--stack lds--stack--nowrap lds--stack--space-lg lds--stack--direction--column lds--stack--align-start lds--stack--justify-start"
  >
    Test
  </div>
</div>
`;

exports[`Stack > should render with wrap = true 1`] = `
<div>
  <div
    class="lds--stack lds--stack--wrap lds--stack--space-lg lds--stack--direction--column lds--stack--align-start lds--stack--justify-start"
  >
    Test
  </div>
</div>
`;
